File: 3517-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int n = s.size();
        vector<int>cnt(26,0);
        for(char c:s){
            cnt[c-'a']++;
        }
        string left ="";
        string mid ="";
        for(int i =0;i<26;i++){
            if(cnt[i]%2==1){
                mid = char(i+'a');
            }
            left += string(cnt[i] / 2, char(i + 'a'));
        }
         string right = left;
        reverse(right.begin(), right.end());

        return left + mid + right;
    }
};

File: 3517-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.py
class Solution(object):
    def smallestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        n = len(s)
        count ={}

        for c in s:
            index = ord(c) - ord('a')
            count[index] = count.get(index, 0) + 1
        left =""
        mid =""
        for i in range (0,26):
            freq = count.get(i,0)

            if freq%2 == 1:
                mid += chr(ord('a')+i)
            left +=  chr(ord('a') + i) * (count.get(i, 0) // 2)
        
        right = left[::-1]

        return left+mid+right